nvdaHelper/local/screenCurtain.cpp

```diff
@@ -0,0 +1,186 @@
+	BITMAP ddScreenshot;
+	HGDIOBJ oldObj;
+	BITMAPINFOHEADER diScreenshotHeader;
+	bool bStatus ;
+	Status sStatus;
+
+	// The desktop window covers the entire virtual screen.
+	desktopWnd = GetDesktopWindow();
+	if (desktopWnd == NULL) {
+		LOG_ERROR(L"Failed to get handle for desktop window.");
+		return false;
+	}
+	wil::unique_hdc_window desktopDc(GetDC(desktopWnd));
+	if (!desktopDc.is_valid()) {
+		LOG_ERROR(L"Failed to get device context for desktop.");
+		return false;
+	}
+	wil::unique_hdc captureDc(CreateCompatibleDC(desktopDc.get()));
+	if (!captureDc.is_valid()) {
+		LOG_ERROR("Failed to create compatible device context.");
+		return false;
+	}
+	wil::unique_hbitmap captureBitmap(CreateCompatibleBitmap(desktopDc.get(), screenWidth, screenHeight));
+	if (captureBitmap == NULL) {
+		LOG_ERROR(L"Failed to create compatible bitmap.");
+		return false;
+	}
+
+	// Set captureDc to draw to captureBitmap.
+	oldObj = SelectObject(captureDc.get(), captureBitmap.get());
+	if (oldObj == NULL) {
+		LOG_ERROR("Failed to select capture bitmap into capture device context.");
+		return false;
+	}
+
+	// Replace the contents of captureDc with those of desktopDc
+	bStatus = BitBlt(
+		// Destination device context
+		captureDc.get(),
+		// Top left of destination
+		0, 0,
+		// Size of source and destination
+		screenWidth, screenHeight,
+		// Source device context
+		desktopDc.get(),
+		// Top left of source
+		screenOriginX, screenOriginY,
+		// Raster operation to perform.
+		// In this case, replace destination with source.
+		SRCCOPY);
+	// Restore captureDC for safety.
+	SelectObject(captureDc.get(), oldObj);
+	if (!bStatus) {
+		LOG_ERROR("Failed to bit blit desktop device context to capture device context. Error #" << GetLastError());
+		return false;
+	}
+
+	// Get properties of captureBitmap
+	bytesWritten = GetObject(captureBitmap.get(), sizeof(BITMAP), &ddScreenshot);
+	if (bytesWritten == 0) {
+		LOG_ERROR("Failed to get bitmap metadata.");
+		return false;
+	}
+
+	// And use them to set the properties of the device independent bitmap.
+	diScreenshotHeader.biSize = sizeof(BITMAPINFOHEADER);
+	diScreenshotHeader.biWidth = ddScreenshot.bmWidth;
+	diScreenshotHeader.biHeight = ddScreenshot.bmHeight;
+	diScreenshotHeader.biPlanes = 1;			  // Can only ever be 1
+	diScreenshotHeader.biBitCount = 32;		  // High byte unused
+	diScreenshotHeader.biCompression = BI_RGB; // Uncompressed
+	diScreenshotHeader.biSizeImage = 0;		  // Unneeded as uncompressed
+	diScreenshotHeader.biXPelsPerMeter = 0;
+	diScreenshotHeader.biYPelsPerMeter = 0;
+	diScreenshotHeader.biClrUsed = 0;
+	diScreenshotHeader.biClrImportant = 0; // All colours are needed
+
+	// Calculate the size (in bytes) of the DIB.
+	diScreenshotSize = ((ddScreenshot.bmWidth * diScreenshotHeader.biBitCount + 31) / 32) * 4 * ddScreenshot.bmHeight;
+
+	try {
+		// Convert the device-dependent bitmap to a device-independent bitmap.
+		auto diScreenshotBits = make_shared<char[]>(diScreenshotSize);
```

Comment:
when I initially worked on this, reef suggest to set some canary bits, to ensure the screen grab converts them to 0